app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -860,6 +864,18 @@ export default Component.extend({
     this.send("togglePreview");
   },
 
+  _isInQuote(element) {
+    for (let e = element; e; e = e.parentElement) {
```

Comment:
if I read this correctly you want to check `element` and `element.parentElement` right?

I think a more readable solution would be:

```javascript
[element, element.parentElement].forEach(e => {
  // code
});
```

Reply:
I think the for loop here go back up the whole tree. Not just one parent 😉

Reply:
yes indeed, hmmm well then ok 🤷‍♂️

Reply:
Such loops implemented with `for` can be confusing. I have this version with `while` more self explainable:

```javascript
  _isInQuote(element) {
    let parent = element.parentElement;
    while (parent && !this._isPreviewRoot(parent)) {
      if (this._isQuote(parent)) {
        return true;
      }

      parent = parent.parentElement;
    }

    return false;
  },

  _isPreviewRoot(element) {
    return (
      element.tagName === "DIV" &&
      element.classList.contains("d-editor-preview")
    );
  },

  _isQuote(element) {
    return element.tagName === "ASIDE" && element.classList.contains("quote");
  },
```